src/tools/tests/schema2kotlin-test.ts

```diff
@@ -219,10 +250,122 @@ describe('schema2kotlin', () => {
         'typealias P_H1_2 = AbstractP.P_H1_2',
       ]
     ));
+    it('Progressively constrained variables', async () => await assertSchemaAliases(
+      `particle T
+         h1: reads ~a
```

Comment:
Could we have another test with more than one type variable, to ensure separate entities are referenced in aliases?

E.g.
```
particle T
  h1: reads ~a
  h2: writes &~a with {x: Number}
  h3: reads ~b
  h4: reads [~b with {a: Text}]
```

the result would be probably:
```
        'typealias T_H1 = AbstractT.TInternal1',
        'typealias T_H2 = AbstractT.TInternal1',
        'typealias T_H3 = AbstractT.TInternal2',
        'typealias T_H4 = AbstractT.TInternal2',
```

Reply:
That's a good idea. This should be proven in schema2graph, but I the idea of another test if only to document the behavior.

Reply:
Surprisingly to me, this test caught a bug! Will investigate now.

Reply:
Patched.

Reply:
Haha, it's always surprising when a test you just wrote actually exposes a problem you didn't know you had! :D happened to me just a few days ago as well, imagine my surprise.